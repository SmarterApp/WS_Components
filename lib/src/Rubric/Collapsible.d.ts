import * as React from "react";
export interface CollapsibleProps {
    className?: string;
    style?: React.CSSProperties;
    label: string;
}
export interface CollapsibleState {
    isCollapsed: boolean;
}
export declare class Collapsible extends React.Component<CollapsibleProps, CollapsibleState> {
    constructor(props: CollapsibleProps);
    toggleCollapsed(): void;
    render(): JSX.Element;
}
