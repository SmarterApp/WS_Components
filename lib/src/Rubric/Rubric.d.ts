import * as React from "react";
import { RubricModel } from "./RubricModels";
export declare class Rubric extends React.Component<RubricModel, {}> {
    renderRubrics(): JSX.Element | undefined;
    renderSampleResponses(): JSX.Element | undefined;
    render(): JSX.Element;
}
