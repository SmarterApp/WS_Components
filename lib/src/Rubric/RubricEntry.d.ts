import * as React from "react";
import { RubricEntryModel } from "./RubricModels";
export declare class RubricEntry extends React.Component<RubricEntryModel, {}> {
    render(): JSX.Element;
}
