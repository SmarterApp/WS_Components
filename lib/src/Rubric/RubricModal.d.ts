import * as React from "react";
import { RubricTableProps } from "@src/index";
export interface RubricModalProps extends RubricTableProps {
    showModal?: boolean;
}
export interface RubricModalState {
    showModal: boolean;
}
export declare class RubricModal extends React.Component<RubricModalProps, RubricModalState> {
    constructor(props: RubricModalProps);
    handleShowModal: () => void;
    handleHideModal: () => void;
    render(): JSX.Element;
}
