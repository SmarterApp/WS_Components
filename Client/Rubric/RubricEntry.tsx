import * as React from 'react';
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as Collapsible from './Collapsible';

export class RubricEntryComponent extends React.Component<AboutItemModels.RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <Collapsible.CComponent label={label}>
                <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
            </Collapsible.CComponent>
        );
    }
}