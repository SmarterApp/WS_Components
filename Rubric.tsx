import * as React from 'react';

export interface SampleResponse {
    purpose: string;
    scorePoint: string;
    name: string;
    sampleContent: string;
}

export interface RubricSample {
    maxValue: string;
    minValue: string;
    sampleResponses: SampleResponse[];
}

export interface RubricEntry {
    scorepoint: string;
    name: string;
    value: string;
}

export interface Rubric {
    language: string;
    rubricEntries: RubricEntry[];
    samples: RubricSample[];
}

export class RubricEntryComponent extends React.Component<RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
        );
    }
}

export class SampleResponseComponent extends React.Component<SampleResponse, {}> {
    render() {
        const pointLabel = this.props.scorePoint == "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorePoint} ${pointLabel})`;
        return (
            <div className="sample-response">
                <b>Purpose: </b> {this.props.purpose} <br />
                <b>Sample Response: </b> <br />
                <div dangerouslySetInnerHTML={{ __html: this.props.sampleContent }} />
            </div>
        );
    }
}

export class RubricComponent extends React.Component<Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }
        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <div className="item-information-content">
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </div>
        );
    }
}