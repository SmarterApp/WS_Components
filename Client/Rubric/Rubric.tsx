import * as React from 'react'
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as RubricEntry from './RubricEntry';
import * as Collapsible from './Collapsible';
import * as SampleResponse from '../SampleResponse/SampleResponse';

export class RubricComponent extends React.Component<AboutItemModels.Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntry.RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
    
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponse.SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }

        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <Collapsible.CComponent label={label}>
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </Collapsible.CComponent>
        );
    }
}