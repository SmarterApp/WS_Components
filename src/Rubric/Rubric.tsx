import * as React from "react";
import { Collapsible } from "./Collapsible";
import { RubricEntry } from "./RubricEntry";
import { RubricModel } from "./RubricModels";
import { SampleResponse } from "./SampleResponse";

export class Rubric extends React.Component<RubricModel, {}> {
  renderRubrics() {
    const rubricEntries = this.props.rubricEntries.map((re, i) => (
      <RubricEntry {...re} key={String(i)} />
    ));
    if (rubricEntries.length === 0) {
      return null;
    }

    const rubrics = (
      <div>
        <h4>Rubrics</h4>
        {rubricEntries}
      </div>
    );
    return rubrics;
  }

  renderSampleResponses() {
    const rubricSamples: JSX.Element[] = [];
    let i: number = 0;

    for (const sample of this.props.samples) {
      const key = `${i}:`;
      const responses = sample.sampleResponses.map((sr, idx) => (
        <SampleResponse {...sr} key={key + String(idx)} />
      ));
      rubricSamples.push(...responses);
      i++;
    }

    if (rubricSamples.length === 0) {
      return null;
    }

    const sampleResponses = (
      <div>
        <h4>Sample Responses</h4>
        {rubricSamples}
      </div>
    );

    return sampleResponses;
  }

  render() {
    const label = `${this.props.language} Rubric`;
    return (
      <Collapsible label={label}>
        {this.renderRubrics()}
        {this.renderSampleResponses()}
      </Collapsible>
    );
  }
}
