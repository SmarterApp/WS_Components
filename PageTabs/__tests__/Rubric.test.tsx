import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { RubricEntryComponent, SampleResponseComponent } from '../Rubric';
import { shallow, mount } from 'enzyme';

describe("RubricEntryComponent", () => {
    const props = {
        scorepoint: "123",
        name:  "Steve Harvey",
        value: "val"
    }

    it("matches snapshot", () => {
        expect(shallow(<RubricEntryComponent {...props}/>)).toMatchSnapshot();;
    })
})


describe("SampleResponseComponent", () => {
    const props = {
        purpose: "purpose",
        scorePoint: "scorePoint",
        name: "name",
        sampleContent: "sampleContent"
    }

    it("matches snapshot", () => {
        expect(shallow(<SampleResponseComponent {...props}/>)).toMatchSnapshot();;
    })
})