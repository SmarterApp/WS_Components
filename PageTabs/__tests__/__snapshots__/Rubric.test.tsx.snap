// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubricEntryComponent matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    dangerouslySetInnerHTML={
        Object {
            "__html": "val",
          }
    }
/>,
  "nodes": Array [
    <div
      dangerouslySetInnerHTML={
            Object {
                  "__html": "val",
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RubricEntryComponent
        name="Steve Harvey"
        scorepoint="123"
        value="val"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RubricEntryComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "name": "Steve Harvey",
          "scorepoint": "123",
          "value": "val",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "val",
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "val",
                            }
          }
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RubricEntryComponent
    name="Steve Harvey"
    scorepoint="123"
    value="val"
/>,
}
`;

exports[`SampleResponseComponent matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sample-response"
>
    <b>
        Purpose: 
    </b>
     
    purpose
     
    <br />
    <b>
        Sample Response: 
    </b>
     
    <br />
    <div
        dangerouslySetInnerHTML={
            Object {
                "__html": "sampleContent",
              }
        }
    />
</div>,
  "nodes": Array [
    <div
      className="sample-response"
>
      <b>
            Purpose: 
      </b>
       
      purpose
       
      <br />
      <b>
            Sample Response: 
      </b>
       
      <br />
      <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "sampleContent",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SampleResponseComponent
        name="name"
        purpose="purpose"
        sampleContent="sampleContent"
        scorePoint="scorePoint"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SampleResponseComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "name": "name",
          "purpose": "purpose",
          "sampleContent": "sampleContent",
          "scorePoint": "scorePoint",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sample-response"
>
          <b>
                    Purpose: 
          </b>
           
          purpose
           
          <br />
          <b>
                    Sample Response: 
          </b>
           
          <br />
          <div
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "sampleContent",
                                      }
                    }
          />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="sample-response"
>
          <b>
                    Purpose: 
          </b>
           
          purpose
           
          <br />
          <b>
                    Sample Response: 
          </b>
           
          <br />
          <div
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": "sampleContent",
                                      }
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SampleResponseComponent
    name="name"
    purpose="purpose"
    sampleContent="sampleContent"
    scorePoint="scorePoint"
/>,
}
`;

exports[`rubric component matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="item-information-content"
>
    <div>
        <h4>
            Rubrics
        </h4>
        <RubricEntryComponent
            name="Steve Harvey"
            scorepoint="123"
            value="val"
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="item-information-content"
>
      <div>
            <h4>
                  Rubrics
            </h4>
            <RubricEntryComponent
                  name="Steve Harvey"
                  scorepoint="123"
                  value="val"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RubricComponent
        language="english"
        rubricEntries={
                Array [
                        Object {
                          "name": "Steve Harvey",
                          "scorepoint": "123",
                          "value": "val",
                        },
                      ]
        }
        samples={Array []}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RubricComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "language": "english",
          "rubricEntries": Array [
            Object {
              "name": "Steve Harvey",
              "scorepoint": "123",
              "value": "val",
            },
          ],
          "samples": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="item-information-content"
>
          <div>
                    <h4>
                              Rubrics
                    </h4>
                    <RubricEntryComponent
                              name="Steve Harvey"
                              scorepoint="123"
                              value="val"
                    />
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="item-information-content"
>
          <div>
                    <h4>
                              Rubrics
                    </h4>
                    <RubricEntryComponent
                              name="Steve Harvey"
                              scorepoint="123"
                              value="val"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RubricComponent
    language="english"
    rubricEntries={
        Array [
            Object {
              "name": "Steve Harvey",
              "scorepoint": "123",
              "value": "val",
            },
          ]
    }
    samples={Array []}
/>,
}
`;
